src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs

```diff
@@ -180,19 +184,48 @@ private void ExpandCategory(TreeViewItem sender, NodeCategoryViewModel selectedC
             foreach (var categoryToBeCollapsed in allExpandedCategories)
             {
                 // If class button was clicked.
-                if (categoryToBeExpanded == null)
+                if (selectedClass.SubCategories.Count == 0)
```

Comment:
Should we check for `if (selectedClass is ClassesNodeCategoryViewModel)`? That way we are sure it is a class button that the user clicks on.